test/tla/specs/checker/BasicChecker.tla

```diff
@@ -0,0 +1,111 @@
+-------------------------- MODULE BasicChecker --------------------------------
+(*
+ * A basic specification of the exploration algorithm that is implemented in
+ * Apalache. We focus on the algorithm itself and omit the non-essential parts.
+ * Hence, we make the following assumptions:
+ *
+ * - The set of states and the transition relation are predefined.
+ * 
+ * - For simplicity, we consider all states to be encoded with integers,
+ * instead of considering complex data structures.
+ *
+ * Igor Konnov, Informal Systems, 2022
+ *)
+
+EXTENDS Integers, Sequences, Apalache, Solver_typedefs
+
+CONSTANTS
+    \* The set of all potential states.
+    \*
+    \* @type: Set(Int);
+    STATES,
+    \* The set of the initial states.
+    \*
+    \* @type: Set(Int);
+    INIT_STATES,
+    \* The set of all potential transitions.
+    \*
+    \* @type: Set(<<Int, Int>>);
+    TRANSITIONS,
+    \* The invariant to be checked.
+    \*
+    \* @type: Set(Int);
+    INVARIANT,
+    \* The maximal number of steps to consider.
+    \*
+    \* @type: Int;
+    MAX_STEPS
+
+VARIABLES
+    \* The state of the search.
+    \*
+    \* @type: { code: Str, trace: Seq(Int) };
+    searchState,
+    \* The current step.
+    \*
+    \* @type: Int;
+    stepNo,
+    \* The solver state
+    \*
+    \* @type: CONTEXT;
+    context
+
+INSTANCE Solver    
+
+TIME_FRAMES == 1..MAX_STEPS + 1
+
+CheckStep(ctx, trace) ==
+    IF ~IsModel(ctx, trace)
+    THEN [ code |-> "Deadlock", trace |-> <<>> ]
+    ELSE LET withNotInv == InsertPred(ctx, STATES \ INVARIANT) IN
+         IF IsModel(withNotInv, trace)
+         THEN [ code |-> "Error", trace |-> trace ]
+         ELSE [ code |-> "NoError", trace |-> <<>> ]
+
+Init ==
+    /\ stepNo = 0
+    /\ context = ContextPush(ContextEmpty, { INIT_STATES }, {})
+    /\ \E trace \in [ { 1 } -> STATES ]:
+        LET traceSeq == FunAsSeq(trace, 1, 1) IN
+        searchState = CheckStep(context, traceSeq)
+
+Next ==
+    /\ searchState.code = "NoError"
+    /\ stepNo < MAX_STEPS
+    /\ stepNo' = stepNo + 1
+    /\ \E trace \in [ TIME_FRAMES -> STATES ]:
+        LET traceSeq == FunAsSeq(trace, stepNo + 2, MAX_STEPS + 1) IN
+        LET newCtx == ContextPush(context, {}, { TRANSITIONS }) IN
```

Comment:
If we only ever push a singleton transition set, why is it modeled as set of sets in the solver context?